### ISSUE TITLE
[Nullable ref types] Infer non-nullablity with array `?.Length` check

### ISSUE THREAD

**Issue by garuma** [dotnet/roslyn#31639]
**Version Used**: VisualStudio.16.int.d16.0stg/16.0.0-pre.2.0+28408.101.d16.0stg
C# Tools   2.11.0-beta2-63529-05+66aa49da81717aa54090a9e94c2f575ddb1959bb

**Steps to Reproduce**:

Take following code block:

``` csharp
protected GreenNode(SyntaxKind kind, int fullWidth, DiagnosticInfo[]? diagnostics, SyntaxAnnotation[]? annotations)
    : this(kind, fullWidth)
{
    if (diagnostics?.Length > 0)
    {
        this.flags |= NodeFlags.ContainsDiagnostics;
        diagnosticsTable.Add(this, diagnostics);
    }
    if (annotations?.Length > 0)
    {
        this.flags |= NodeFlags.ContainsAnnotations;
        annotationsTable.Add(this, annotations);
    }
}
```
Currently the parameters to `Add` (both `diagnostics` and `annotations`) are flagged as "Possible null reference argument for parameter" however because of the form of the if check it should be possible to infer that those arguments are non-null at that stage.

**Comment by jaredpar** [comment]
Thanks for reporting. This looks like a dupe of #29956
